pkg/provisioner/cloudprovisioner.go

```diff
@@ -743,6 +743,9 @@ func (c *CloudProvisioner) validateCreateVolumeRequestParams(
 	capacityRange *v1alpha1.CapacityRange,
 	volumeCaps []v1alpha1.VolumeCapability,
 	params map[string]string) error {
+	if capacityRange == nil {
```

Comment:
If we want to omit this check, we need to make CapacityRange required. It is now optional.

Reply:
@landreasyan  You are correct, as per the definition of `capacityRange ` in the CSI types, the capacityRange is optional. 

However, I did notice that the implementation of controllerserver in v1 also considers the capacityRange to be [non `nil`](https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/09a026cd096988623d652d6c683e03c9beaeb5ea/pkg/azuredisk/controllerserver.go#L106).

To confirm this behavior, I modified the `TestCreateVolume ` unit test cases for v1 controllerserver by removing the capacityRange in the `valid request` [test case](https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/09a026cd096988623d652d6c683e03c9beaeb5ea/pkg/azuredisk/controllerserver_test.go#L457) and it did fail on the same error too.  

I'm investigating further and will post an update when I find something definite.

Reply:
I have updated the handling for `capacityRange == nil` and defaulted the required size to MinimumDiskSize